solidity/random-beacon/README.adoc

```diff
@@ -177,103 +166,122 @@ When the hard timeout is reached, anyone can notify about this fact by calling
 
 [[inactivity]]
 === Inactivity notification
 
 Off-chain clients are free to execute any heartbeat protocol they want to ensure
-group members are alive and nodes are operating properly.
+group member key material is still available and nodes are operating properly.
 
 [TIP]
 One example of a heartbeat protocol is signing some piece of information every
-nth blocks and first making sure the information cannot be used for
-`RandomBeacon.reportUnauthorizedSigning()`, that is, the signed information can
-not become `msg.sender` for `reportUnauthorizedSigning` call.
+nth block and making sure this piece of information cannot be used for
+`RandomBeacon.reportUnauthorizedSigning()`. Specifically, the signed piece of
+information can not become `msg.sender` for `reportUnauthorizedSigning` call.
 
-Group members can agree upon members who are permanently inactive and issue an
-operator inactivity claim. If the required threshold of group members signed
+Group members can agree to punish members who are permanently inactive and issue
+an operator inactivity claim. If the required threshold of group members signed
 the operator inactivity claim, they can submit it to
-`RandomBeacon.notifyOperatorInactivity(Inactivity.Claim calldata claim, uint256 nonce)`
+`RandomBeacon.notifyOperatorInactivity(Inactivity.Claim calldata claim, uint256 nonce, int32[] calldata groupMembers)`
 function and have the group members who are inactive excluded from
 the sortition pool rewards for a governable time period.
 
 This approach is theoretically susceptible to group members colluding together
 but because a reasonably high number of operators is needed to sign a claim and
-operators signing the claim other than the submitter receive nothing in return,
+operators signing the claim receive nothing in return,
 we consider this approach safe and good enough. An important advantage of this
 approach is that honest players can decide off-chain when it makes sense to
 submit an operator inactivity claim and mark someone as ineligible for rewards.
 For example, marking an operator ineligible for rewards for the next two weeks
 have a higher impact than prolonging reward ineligibility for 10 minutes for an
 operator that was already marked as ineligible for rewards. This approach does
 not increase the gas cost of a happy path and leaves some freedom to group
-members. They may mark as ineligible operators who turned off their nodes,
+members. They can mark as ineligible operators who turned off their nodes,
 operators whose nodes never participate in signing because they are
 misconfigured, or operators who notoriously miss their turn in submitting relay
-entry.
+entries.
+
+=== Rewards
+
+T rewards are allocated to all operators registered in the beacon sortition
+pool, excluding operators who were marked as ineligible for rewards as a result
+of <<inactivity, inactivity>>. Rewards are allocated proportionally to the
```

Comment:
The `[[inactivity]]` section describes only the situation when the `notifyOperatorInactivity()` function is called. The operators can be marked as ineligible for rewards also as a result of inactivity during off-chain DKG, when `approveDkgResult()` function is called, we should includa that case here.

Reply:
07ad25c3b3bf9c7f2a23853a0b071397ee62a873